src/canvas-force-graph.js

```diff
@@ -196,35 +207,38 @@ export default Kapsule({
         beforeCustomLinks.forEach(link => state.linkCanvasObject(link, ctx, state.globalScale));
         ctx.restore();
 
-        // Bundle strokes per unique color/width for performance optimization
-        const linksPerColor = indexBy(defaultPaintLinks, [getColor, getWidth]);
+        // Bundle strokes per unique color/width/dash for performance optimization
+        const linksPerColor = indexBy(defaultPaintLinks, [getColor, getWidth, getLineDash]);
```

Comment:
Unfortunately this indexing won't work because it would be using the lineDash array references, which will be different for arrays with the same values.

Reply:
I think indexBy stringifies arrays, which is why I have the typeof string type check in `parseLineDashSegments` below

Reply:
Confirmed in devtools that `indexBy` is stringifying arrays with `[]` values coerced to empty strings and that the totals add up to the 79 links created in the dash-odd-links example. So in the example, repeated evaluations of `[5, 5]` from different links cause them all to be put into the `"5,5"` "bucket"

![2020-03-26 20_30_07-DevTools - localhost_8080_example_dash-odd-links_](https://user-images.githubusercontent.com/563860/77631613-e8d5d600-6fa0-11ea-92cb-559faca5efc5.png)

So with this assurance, the function `parseLineDashSegments` has been removed and the call to it simplified to just the raw `split`/`parseFloat` expression